projects/FHC-AtmelStudio/Fussl-01/Fussl-01/f_range_int.h

```diff
@@ -0,0 +1,153 @@
+/*
+* f_range_int.h
+*
+* Created: 13.03.2018 17:52:31
+* @author Maximilian Starke
+*/
+
+#ifndef __F_RANGE_INT_H__
+#define __F_RANGE_INT_H__
+namespace fsl {
+	namespace ver_1_0 {
+		namespace lg {
+			
+			/*! @brief Type for Integers in a range {\a base_type_constants::ZERO, ... , \a base_type_constants::MAX}
+			@details Stores a number in {\a base_type_constants::ZERO, ... , \a base_type_constants::MAX} or #OUT_OF_RANGE.
+			@tparam _base_type The base integer type that is used internally. !add description, what this type must bring with it. Must provide a total order with comparison operator < and operator ==. ??It must provide constructors for _base_type(0), _base_type(1), _base_type(-1)
+			@tparam _RANGE The maximum value that a #fsl::ver_1_0::lg::range_int can take + 1. Must be positive.
+			@tparam _OVERFLOW_RESULTS_IN_OUT_OF_RANGE Defines whether some arithmetic operation (e.g. #operator+) leading to an overflow will result in #OUT_OF_RANGE or if it will behave like a ring in mathematics.
+			@tparam _OUT_OF_RANGE_IS_ABSORBING If it is set true any arithmetical operation will result in #OUT_OF_RANGE if one of the operands is #OUT_OF_RANGE.
+			*/
+			template <typename _base_type, _base_type _RANGE, bool _OVERFLOW_RESULTS_IN_OUT_OF_RANGE = true, bool _OUT_OF_RANGE_IS_ABSORBING = true>
```

Comment:
Make sure that all requirements to the template parameters are listed in this doc block!

Reply:
I added all requirements that I could see, hope I did not miss one